### PR TITLE
Support wildcard domain pattern in analytics allowed hostnames

### PR DESCRIPTION
**Pull request by devkiran**
## Summary by CodeRabbit

- **New Features**
  - Added support for wildcard subdomains (e.g., *.example.com) in allowed hostname settings and validation.
  - Updated input placeholder to indicate wildcard subdomains are permitted.

- **Bug Fixes**
  - Improved hostname validation logic for more accurate and consistent results.

- **Tests**
  - Introduced comprehensive tests to ensure correct behavior of hostname validation, including wildcard and edge cases.

### CODE REVIEW

**Comment by devkiran** [comment]
@coderabbitai any breaking changes?

File: apps/web/lib/analytics/verify-analytics-allowed-hostnames.ts [hunk 1]
```diff
@@ -5,13 +5,45 @@ export const verifyAnalyticsAllowedHostnames = ({
...
+    return true;
   }
 
-  return true;
+  // Check for wildcard subdomain matches
+  const wildcardMatches = allowedHostnames
+    .filter((domain) => domain.startsWith("*."))
+    .map((domain) => domain.slice(2)); // Remove the "*.", leaving just the domain
+
+  for (const domain of wildcardMatches) {
```
Comment (steven-tey) [inline apps/web/lib/analytics/verify-analytics-allowed-hostnames.ts:34]: @coderabbit won't this allow potential malicious actors to use something like `testdomain.com` to spoof `domain.com` as well?

Comment (steven-tey) [reply apps/web/lib/analytics/verify-analytics-allowed-hostnames.ts:34]: Yes please 🙏

Comment (steven-tey) [reply apps/web/lib/analytics/verify-analytics-allowed-hostnames.ts:34]: @coderabbit need your help with the code changes, please come back from your burrow 🥺

**Comment by devkiran** [comment]
@coderabbitai final review please

**Review by steven-tey — APPROVED** [review]
w
